webapp-django/crashstats/symbols/views.py

```diff
@@ -48,6 +49,80 @@ def check_symbols_archive_content(content, header_length=2):
                 )
 
 
+def unpack_and_upload(iterator, symbols_upload, bucket_name, bucket_location):
+    necessary_setting_keys = (
+        'AWS_ACCESS_KEY',
+        'AWS_SECRET_ACCESS_KEY',
+        'SYMBOLS_BUCKET_DEFAULT_LOCATION',
+        'SYMBOLS_BUCKET_DEFAULT_NAME',
+        'SYMBOLS_FILE_PREFIX',
+    )
+    for key in necessary_setting_keys:
+        if not getattr(settings, key):
+            raise ImproperlyConfigured(
+                "Setting %s must be set" % key
+            )
+
+    conn = boto.connect_s3(
+        settings.AWS_ACCESS_KEY,
+        settings.AWS_SECRET_ACCESS_KEY
+    )
+    assert bucket_name
+    try:
+        bucket = conn.get_bucket(bucket_name)
+    except boto.exception.S3ResponseError:
+        bucket = conn.create_bucket(bucket_name, bucket_location)
+
+    total_uploaded = 0
+    for member in iterator:
+        key_name = os.path.join(
+            settings.SYMBOLS_FILE_PREFIX, member.name
+        )
+        key = bucket.get_key(key_name)
+
+        # let's assume first that we need to add a new key
+        prefix = "+"
+        if key:
+            # key already exists, but is it the same size?
+            if key.size != member.size:
+                # we're changing our minds, so it get uploaded again
```

Comment:
this "changing our minds" comment is confusing.. maybe just ```# file size in S3 is different, upload the new one```?
If you disagree at least ```s/get/gets/``` :)